packages/client-python/src/rocketride/mixins/execution.py

```
@@ -254,14 +254,17 @@ async def use(
-                pipeline_config = parsed.get('pipeline', parsed) if isinstance(parsed, dict) else parsed
+            try:
+                with open(filepath, 'r', encoding='utf-8') as file:
+                    # Prefer JSON5 for better developer experience (comments, trailing commas)
+                    if json5:
+                        parsed = json5.load(file)
+                    else:
+                        parsed = json.load(file)
+                    # .pipe files wrap the config in { "pipeline": { ... } } — unwrap if present
+                    pipeline_config = parsed.get('pipeline', parsed) if isinstance(parsed, dict) else parsed
+            except FileNotFoundError:
+                raise FileNotFoundError(f"Pipeline file not found: '{filepath}'. Please provide a valid file path or use inline pipeline configuration.")
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)